### PR TITLE
Back and forground index export

### PR DESCRIPTION
**Pull request by LokiMidgard**
This is an proposal to store background and foreground color index.

I've added two section `screen` and `popup`. Both sections contain two keys `FORGROUND` and `BACKGROUND`. The assignable values are keys of the `table` section.

**sample**
```ini
[table]
DARK_BLACK = 12,12,12
DARK_BLUE = 0,55,218
DARK_GREEN = 19,161,14
DARK_CYAN = 58,150,221
DARK_RED = 197,15,31
DARK_MAGENTA = 1,36,86
DARK_YELLOW = 238,237,240
DARK_WHITE = 204,204,204
BRIGHT_BLACK = 118,118,118
BRIGHT_BLUE = 59,120,255
BRIGHT_GREEN = 22,198,12
BRIGHT_CYAN = 97,214,214
BRIGHT_RED = 231,72,86
BRIGHT_MAGENTA = 180,0,158
BRIGHT_YELLOW = 249,241,165
BRIGHT_WHITE = 242,242,242

[screen]
FOREGROUND = DARK_YELLOW
BACKGROUND = DARK_MAGENTA

[popup]
FOREGROUND = DARK_CYAN
BACKGROUND = BRIGHT_WHITE

[info]
name = powershell-default
```

### CODE REVIEW

**Comment by zadjii-msft** [comment]
I'm okay with this for the time being.

Ideally, I'd want to also have a way to export the "terminal" default foreground and background colors, but since those aren't really query-able, nor set-able with VT, then it doesn't really make sense to include them in the scheme currently. We can come back and add support to colortool for those later, once the support is there in the console itself.

File: tools/ColorTool/ColorTool/IniSchemeParser.cs [hunk 1]
```diff
@@ -85,6 +86,10 @@ public ColorScheme ParseScheme(string schemeName, bool reportErrors = false)
 
             string[] tableStrings = new string[COLOR_TABLE_SIZE];
             uint[] colorTable = null;
+            uint? forgroundColor = null;
```
Comment (miniksa) [inline tools/ColorTool/ColorTool/IniSchemeParser.cs]: Can you fix the typo here? It should be "foregroundColor" here and for all references (as well as "popupForegroundColor" below).

File: tools/ColorTool/ColorTool/IniSchemeParser.cs [hunk 2]
```diff
@@ -127,12 +154,33 @@ public ColorScheme ParseScheme(string schemeName, bool reportErrors = false)
 
             if (colorTable != null)
             {
-                return new ColorScheme { colorTable = colorTable };
+                return new ColorScheme { colorTable = colorTable, background = backgroundColor, foreground = forgroundColor, popupBackground = popupBackgroundColor, popupForeground = popupForgroundColor };
```
Comment (miniksa) [inline tools/ColorTool/ColorTool/IniSchemeParser.cs]: I feel like this is a lot of parameters to add. Can they be put in a struct and passed that way as a single unit?

File: tools/ColorTool/ColorTool/IniSchemeParser.cs [hunk 3]
```diff
@@ -127,12 +154,33 @@ public ColorScheme ParseScheme(string schemeName, bool reportErrors = false)
...
             }
             else
             {
                 return null;
             }
+
+            bool ReadAttributes(string section, out string forground, out string background)
```
Comment (miniksa) [inline tools/ColorTool/ColorTool/IniSchemeParser.cs]: here's some more "forground"s that should probably have the "e" added back. :)

File: tools/ColorTool/ColorTool/IniSchemeParser.cs (hunk 2 above)
Comment (LokiMidgard) [reply tools/ColorTool/ColorTool/IniSchemeParser.cs]: Should that struct ony contain the foreground and background colors or should it also contain the color table?

Comment (miniksa) [reply tools/ColorTool/ColorTool/IniSchemeParser.cs]: I feel like conceptually the foreground/background stuff all goes together in one struct and the color table is already one. I'm happy with two parameters here. @zadjii-msft, opinion? Am I just being too picky?

**Comment by LokiMidgard** [comment]
I think I got all "forground". I also added the popup attributes to the JsonParser

**Pull request merged**